drizzlepac/devutils/search_skyframes.py

```diff
@@ -0,0 +1,316 @@
+        specified, no sorting will be performed on query results.
+
+    output_filename : str, optional
+        Optional name of an output .csv file to write the query results to.
+
+    Returns
+    -------
+    results : Pandas DataFrame class
+        A Pandas DataFrame containing the query results.
+    """
+    dataframe = pd.read_csv(master_observations_file, header=0, index_col=0)
+    results = dataframe.query(search_string, engine='python')
+    results = augment_results(results)
+    if date_range:
+        search_string = "{} and '{}' <= dateobs <= '{}'".format(search_string, date_range[0], date_range[1])
+        results = results[(results['dateobs'] >= date_range[0]) & (results['dateobs'] <= date_range[1])]
+    ret_results = results.copy()
+    if output_sorting:
+        results = results.sort_values(by=output_sorting)
+    if output_columns:
+        results = results[output_columns]
+    n_lines = results.index.size
+    print("\n")
+    print(results.to_string())
+    print("\n")
+    if n_lines == 1:
+        print('Query "{}" found 1 result.'.format(search_string))
+    else:
+        print('Query "{}" found {} results.'.format(search_string, n_lines))
+    if output_sorting:
+        col_sorting_str = ", ".join(output_sorting)
+    else:
+        col_sorting_str = "No sorting"
+    print("Column sorting order: {}".format(col_sorting_str))
+    if output_filename:
+        results.to_csv(output_filename)
+        print("Wrote query results to {}".format(output_filename))
+
+    return ret_results
+
+# ------------------------------------------------------------------------------------------------------------
+
+
+def visualize_footprints(results):
+    """Visualize footprints of skycells and exposures in query result
+
+    Parameters
+    ----------
+    results : Pandas DataFrame class
+        Pandas DataFrame containing query results.
+
+    Returns
+    -------
+    Nothing.
+    """
+    # Get list of all the unique skycells in the query results
+    unique_skycell_list = []
+    for idx in results.index:
+        unique_skycell_list.append(results.skycell[idx])
+    unique_skycell_list = list(set(unique_skycell_list))  # remove duplicate items from skycell list
+    n_unique_skycells = len(unique_skycell_list)
+    if n_unique_skycells > 1:
+        add_an_s = "s"
+    else:
+        add_an_s = ""
+    print("Building footprint {} image{}. Please be patient. This may take some time...".format(n_unique_skycells, add_an_s))
+    for skycell_name in unique_skycell_list:
+        skycell = cell_utils.SkyCell.from_name("skycell-{}".format(skycell_name))
+        footprint = cell_utils.SkyFootprint(meta_wcs=skycell.wcs)
+        img_list = results.query('skycell == "{}"'.format(skycell_name)).filename.values.tolist()
+        footprint.build(img_list)
+        footprint_imgname = "skycell-{}-footprint.fits".format(skycell_name)
```

Comment:
It appears that footprints of different instrument/detector combinations for the same SkyCell, let alone filter combinations, would end up overwriting each other with a name that only includes the SkyCell ID.  If the `skycell_name` includes these other aspects as part of the name, that would be fine, but some comments would go a long way towards making that clear.  Otherwise, this name will need to be expanded based on the search criteria.

Reply:
Issue resolved by commit [44205e0](https://github.com/spacetelescope/drizzlepac/pull/964/commits/44205e0991622893e87586fea26e1381a5ee9d24)